### PR TITLE
Asynchronous Ledger Interactions with Server

### PR DESCRIPTION
**Pull request by howard-stearns**
Talks to server rather than using in-memory session data.
Also cleans up browser profiles so that tablet and hud browsers use the same profile.

For internal testing:

- Please test in all four of Desktop/HMD x HUD/Tablet
- Set "inspectionMode": true, in Interface.json
- Goto MARKETPLACE and search for 'flare'. There should be two items (Flaregun and Test Flaregun).
- Make sure other marketplaces (e.g., See All Markets -> Clara.io Explore), embedded browsers (e.g., GOTO -> info and PAL -> connections -> info), and ctrl-b browsers all work as expected.
- Check out https://highfidelity.com/admin and make note of what accounts and balances are present. (Screen shot might be convenient.)
- If you need to retest (or if you are hifiDave), delete the accounts and balance created in earlier sessions, and delete AppData/High Fidelity - PR whatever/Interface/hifikey
  (If you are hifiDave, it's probably easist to use a different account for testing this, so that the creator of the Test Flaregun is not you.
   Eventually, you will want to know that your public key for that is begins: MIIBCgKCAQEA2pRBwp/+Y7OYFZ0c0b3mwBqTdRjIdd8bfn05U8YJWj20KUNVDayg)
- Make your keys. Got to MARKETPLACE -> Inventory and pick a picture. Your balance will be zero, but that's a separate bug. Press the Back button.
- Check your inventory and balance by going to MARKETPLACE -> Inventory (again). Your balance should be 100 and no inventory.
- Buy the Test Flaregun. It should work and be in-world.
- Check your inventory and balance again. It should be 0 and have one Test Flaregun. Clicking the Test Flaregun link should bring up its page.
- Try to buy the Test Flaregun again. The confirmation buy button should be disabled and read Already Owned.
- Echeck out the admin page again: You should have a balance of 0 hfc and 1 090d21c-.... Dave should have 100 more hfc than before.

- Unset "inspectionMode" in Inerface.json.
- Goto MARKETPLACE and search for 'flare'. There should be just one item (Flaregun and _no_ Test Flaregun). There should be no "Inventory" button at the top.

### CODE REVIEW

File: scripts/system/html/js/marketplacesInject.js [hunk 1]
```diff
@@ -114,7 +114,7 @@
             itemId: id,
             itemName: name,
             itemAuthor: author,
-            itemPrice: Math.round(Math.random() * 50),
+            itemPrice: price ? parseInt(price, 10) : Math.round(Math.random() * 50),
```
Comment (zfox23) [inline scripts/system/html/js/marketplacesInject.js]: Is there any reason to keep the "random" logic in here?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2608/HighFidelity-Beta-PR11212-29b717e22d98d2f494b50dbc5a899a3d49888424.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2608/HighFidelity-Beta-PR11212-29b717e22d98d2f494b50dbc5a899a3d49888424.exe

File: interface/src/commerce/Ledger.cpp [hunk 2]
```diff
@@ -10,72 +10,100 @@
...
+// balance answers {status: 'success', data: {balance: integer}}
+// buy and receive_at answer {status: 'success'}
+
+QJsonObject Ledger::apiResponse(const QString& label, QNetworkReply& reply) {
+    QByteArray response = reply.readAll();
+    QJsonObject data = QJsonDocument::fromJson(response).object();
+    qInfo(commerce) << label << "response" << QJsonDocument(data).toJson(QJsonDocument::Compact);
+    return data;
+}
+// Non-200 responses are not json:
```
Comment (davidkelly) [inline interface/src/commerce/Ledger.cpp]: really?  I thought that (for instance) a response when you are not logged in was json, within the /api/v1/commerce stuff anyways.

**Comment by zfox23** [comment]
Trying to buy a free item results in this right now:
image

I know this will change when the checkout UX changes. Noting for posterity.

File: interface/src/commerce/Ledger.cpp (hunk 2 above)
Comment (davidkelly) [inline interface/src/commerce/Ledger.cpp]: well I guess others (like routing errors) are html.

File: scripts/system/html/js/marketplacesInject.js (hunk 1 above)
Comment (howard-stearns) [inline scripts/system/html/js/marketplacesInject.js]: It's kept only to minimize the changes to functionality in this version.
I have no objection to you changing the functionality along with your various UI changes.

File: interface/src/commerce/Ledger.cpp (hunk 2 above)
Comment (howard-stearns) [inline interface/src/commerce/Ledger.cpp]: Yeah, I guess I don't really know. I suspect it's not a clean answer. 
For example, I know for certain that a 404 has an html response.
I could imagine a 401 having a json response for these endpoint, and not for others.
As long as some responses are not json, I guess I can't count on finding the {message: "whatever"}, so show the whole thing.  But if that turns out to not the right thing in practice, then I have no objection to changing as soon as we determine what the right thing is.

**Comment by howard-stearns** [comment]
Someone told me it was tested.

**Pull request merged**